raiden/ui/cli.py

```diff
@@ -308,9 +311,31 @@ def app(address,
         print(e.message)
         sys.exit(1)
 
+    discovery_gasprice = GAS_PRICE
+    discovery_startgas = GAS_LIMIT
+    discovery_tx_cost = discovery_gasprice * discovery_startgas
+
+    while True:
```

Comment:
We need the account data to initialize the blockchain service and we need the service to check for the balance, so I think doing the balance check after account selection is the only easy way.

Usability doesn't suffer too much from it.

Other than that, I did some minor refactoring.